src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensEditsHandler.cs

```diff
@@ -0,0 +1,376 @@
+            => new LSP.SemanticTokensEdit
+            {
+                Start = start,
+                DeleteCount = deleteCount,
+                Data = data
+            };
+
+        private sealed class LongestCommonSemanticTokensSubsequence : LongestCommonSubsequence<SemanticToken[]>
+        {
+            private static readonly LongestCommonSemanticTokensSubsequence s_instance = new LongestCommonSemanticTokensSubsequence();
+
+            protected override bool ItemsEqual(
+                SemanticToken[] oldSemanticTokens, int oldIndex,
+                SemanticToken[] newSemanticTokens, int newIndex)
+                => oldSemanticTokens[oldIndex].Equals(newSemanticTokens[newIndex]);
+
+            public static IEnumerable<SequenceEdit> GetEdits(
+                SemanticToken[] oldSemanticTokens, SemanticToken[] newSemanticTokens)
+                => s_instance.GetEdits(oldSemanticTokens, oldSemanticTokens.Length, newSemanticTokens, newSemanticTokens.Length);
+        }
+
+        /// <summary>
+        /// Stores the values that make up the LSP representation of an individual semantic token.
+        /// </summary>
+#pragma warning disable CA1067 // Override Object.Equals(object) when implementing IEquatable<T>
+        private readonly struct SemanticToken : IEquatable<SemanticToken>
+#pragma warning restore CA1067 // Override Object.Equals(object) when implementing IEquatable<T>
+        {
+            private readonly int _deltaLine;
+            private readonly int _deltaStartCharacter;
+            private readonly int _length;
+            private readonly int _tokenType;
+            private readonly int _tokenModifiers;
+
+            public SemanticToken(int deltaLine, int deltaStartCharacter, int length, int tokenType, int tokenModifiers)
+            {
+                _deltaLine = deltaLine;
+                _deltaStartCharacter = deltaStartCharacter;
+                _length = length;
+                _tokenType = tokenType;
+                _tokenModifiers = tokenModifiers;
+            }
+
+            public void AddToEnd(ArrayBuilder<int> tokensToInsert)
+            {
+                tokensToInsert.Add(_deltaLine);
+                tokensToInsert.Add(_deltaStartCharacter);
+                tokensToInsert.Add(_length);
+                tokensToInsert.Add(_tokenType);
+                tokensToInsert.Add(_tokenModifiers);
+            }
+
+            public bool Equals([AllowNull] SemanticToken otherToken)
+            {
+                return _deltaLine == otherToken._deltaLine &&
+                    _deltaStartCharacter == otherToken._deltaStartCharacter &&
+                    _length == otherToken._length &&
+                    _tokenType == otherToken._tokenType &&
+                    _tokenModifiers == otherToken._tokenModifiers;
+            }
+        }
+
+        private enum SemanticTokenEditKind
+        {
+            None = 0,
```

Comment:
nit: you have a 0 value, but check for `default` in some comparisons.  can you check against None instead.